### PR TITLE
Stork-custom loadBalancer/serviceDiscovery scenario

### PR DESCRIPTION
**Pull request by pjgg**
### Summary

This PR covers a basic implementation of a custom stork LB.

Please select the relevant options.

- [X] New scenario (non-breaking change which adds functionality)
- [X] This change requires a documentation update

### Checklist:
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            .withProperty("stork.pong.load-balancer", "simple")
+            .withProperty("stork.pong.service-discovery", "simple")
+            .withProperty("stork.pong-replica.load-balancer", "simple")
+            .withProperty("stork.pong-replica.service-discovery", "simple")
+            .withProperty("stork.pong.service-discovery.pongServicePort", () -> "" + pongService.getPort())
+            .withProperty("stork.pong-replica.service-discovery.pongReplicaServicePort",
+                    () -> "" + pongReplicaService.getPort());
+
+    @Test
+    public void storkCustomLoadBalancer() {
```
Comment (jsmrcka) [inline service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java:50]: It also tests the custom service discovery, so maybe we should reflect that in the names of the method and the class.

File: service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        Map<String, Integer> uniqueResp = new HashMap<>();
+        final int requestAmount = 100;
+        final int randomLbError = (requestAmount / 2) - 20;
```
Comment (jsmrcka) [inline service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java:53]: The `Error` in the name is a little bit confusing. It is, in fact, something like minimum hits, right? Maybe `randomLbMinHits` would be better. WDYT?

Also, just out of curiosity, is the value an estimate, or is it based on something solid?

File: service-discovery/stork-custom/src/main/java/io/quarkus/ts/stork/custom/SimpleLoadBalancerProvider.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package io.quarkus.ts.stork.custom;
+
+import io.smallrye.stork.api.LoadBalancer;
+import io.smallrye.stork.api.ServiceDiscovery;
+import io.smallrye.stork.api.config.LoadBalancerAttribute;
+import io.smallrye.stork.api.config.LoadBalancerType;
+import io.smallrye.stork.spi.LoadBalancerProvider;
+
+@LoadBalancerType("simple")
+@LoadBalancerAttribute(name = "my-attribute", description = "Attribute that alters the behavior of the LoadBalancer", required = false)
```
Comment (jsmrcka) [inline service-discovery/stork-custom/src/main/java/io/quarkus/ts/stork/custom/SimpleLoadBalancerProvider.java:10]: It seems to me that this attribute is never used. It it still usefull here? It influences the contents of the respective `Configuration`, but we do not verify that anywhere. So I suggest to remove it.

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.ValidatableResponse;
+
+import junit.framework.AssertionFailedError;
+
+@QuarkusScenario
+public class StorkCustomLoadBalancerIT {
```
Comment (jsmrcka) [inline service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java:25]: This scenario is part of `QUARKUS-1413` TD, right? I think we should make that clear somehow, perhaps a `@Tag`.

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java (hunk 2 above)
Comment (pjgg) [reply service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java]: Name changes to the suggested one. 
`> just out of curiosity, is the value an estimate, or is it based on something solid?`

No is an estimation, I mean... a random generator should follow a uniform distribution

```
The probabilities for uniform distribution function are simple to calculate due to the simplicity of the function form. Therefore, there are various applications that this distribution can be used for as shown below: hypothesis testing situations, random sampling cases ...
```
So in theory a `java.util.Random` should distribute the load in a "uniform way", so if we have two instances, as is the case, we should have 50 req per node. However, after some test I saw an error of...10%, to us is not relevant... this is just a fake LB that the only thing that is probing is that the custom LB implementation is working. 

The standard round-robin implementation works [perfectly](https://github.com/quarkus-qe/quarkus-test-suite/blob/main/service-discovery/stork/src/test/java/io/quarkus/ts/stork/StorkLoadBalancerIT.java#L72)!

Comment (jsmrcka) [reply service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java]: Cool, thanks for the clarification.

**Review by jsmrcka — APPROVED** [review]

**Pull request merged**